tests/gpflow/kernels/test_kernels.py

```diff
@@ -670,3 +671,78 @@ def test_combination_LMC_kernels() -> None:
     K = kernel(X, full_cov=True)
     assert K.shape == [N, P, N, P]
     np.testing.assert_allclose(K, K1 + K2)
+
+
+def test_concat_inputs_with_latents() -> None:
```

Comment:
I feel like there are definitely missing tests for the `Categorical` kernel (beyond broadcasting), but I'm probably not the person to spot them! Also don't we want a test that actually uses `Categorical` in a model?